hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1279,8 +1279,21 @@ protected void tryUpgrade(HoodieTableMetaClient metaClient, Option<String> insta
     }
   }
 
+  /**
+   * Rolls back the failed delta commits corresponding to the indexing action.
+   * <p>
+   * TODO(HUDI-5733): This should be cleaned up once the proper fix of rollbacks
+   *  in the metadata table is landed.
+   *
+   * @return {@code true} if rollback happens; {@code false} otherwise.
+   */
+  public boolean lazyRollbackFailedIndexing() {
+    return tableServiceClient.lazyRollbackFailedIndexing();
```

Comment:
This API has to be added so that we can reuse part of rollback logic inside `BaseHoodieTableServiceClient`.